### ISSUE TITLE
[Bug]: Error when displaying the tooltip as of Lizmap 3.8

### ISSUE THREAD

**Issue by guenterw** [3liz/lizmap-web-client#4926]
### What is the bug? (in English)

when activating the display of tooltips in Lizmap (3.8, 3.9, 3.10) the error “An error prevented to load the tooltip data” appears.
See https://cx21-8.webgis.biz/index.php/view/map?repository=buergergis8&project=bauleitplanung_3_8

In version 3.7 it works perfectly with identical data
https://cx21-7.webgis.biz/index.php/view/map?repository=buergergis7&project=bauleitplanung_3_7

### Steps to reproduce the issue

![image](https://github.com/user-attachments/assets/71a3f6d8-22cd-404d-9889-bf4888aa3798)

### Versions, safeguards, check summary etc

**Versions :**

* Lizmap Web Client : 3.8.3-pre.8030
* Lizmap plugin : 4.4.4
* QGIS Desktop : 3.34.11
* QGIS Server : 3.34.12
* Py-QGIS-Server : not used
* QGIS Server plugin atlasprint : 3.4.1
* QGIS Server plugin lizmap_server : 2.11.2
* QGIS Server plugin wfsOutputExtension : 1.8.2

<details>
<summary>List of Lizmap Web Client modules :</summary>
<br/>
* altiProfil : 0.5.5
* altiProfilAdmin : 0.5.5
</details>

<details>
<summary>List of safeguards :</summary>
<br/>
* Mode : normal<br/>
* Allow parent folder : yes<br/>
* Number of parent : 2 folder(s)<br/>
* Prevent other drive : no<br/>
* Prevent PG service : yes<br/>
* Prevent PG Auth DB : yes<br/>
* Force PG user&pass : yes<br/>
* Prevent ECW : yes<br/>
</details>

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### Operating system

Ubuntu 24.04. (Server), Win10

### Browsers

Chrome

### Browsers version

130.0.6723.92

**Comment by Gustry** [comment]
Just quickly looking on GitHub, the error 

> An error prevented to load the tooltip data

comes from `tooltip.loading.error`

https://github.com/3liz/lizmap-web-client/blob/fc239c68428074326f7cacb983f9fc4f1b50707e/assets/src/modules/Tooltip.js#L118

About this code, the message must have a higher warning : 

>  console.log(`Tooltip layer '${layerName}' could not be loaded.`);

Any error in your QGIS Server logs ?

**Comment by guenterw** [comment]
No, there are no entries in the QGIS error log. Neither in the Lizmap error log (backend).
But in the Apache error.log there is close to this entry:
QFont::fromString: Invalid description '(empty)'

The error also occurs in V3.9 and 3.10

**Comment by Antoviscomi** [comment]
@Gustry @guenterw I confirm this behaviour even in mine 3.8.3 LWC version

![Schermata del 2024-11-05 13-21-34](https://github.com/user-attachments/assets/faf6fad1-2ace-45e8-bd49-8a592e8b0124)

**Comment by Gustry** [comment]
I'm not sure what is happening, but it works with the test data : https://sandbox.lizmap.com/lizmap_3_8/index.php/view/map?repository=tests&project=tooltip

**Comment by Gustry** [comment]
I think the issue is when you use the first tab in the plugin when you set the tooltips.
No error occurs if you use the HTML template. Someone can confirm ?

**Comment by Antoviscomi** [comment]
> I think the issue is when you use the first tab in the plugin when you set the tooltips.
> No error occurs if you use the HTML template. Someone can confirm ?

@Gustry here I noticed that tooltips error it come only at map startup, if I close it (error modal popup, as selection dropdown popup) the tooltips tool end to working. The error come out again after a page refresh, and no error in html template but, in template mode there are no contents in tooltips modal popup (in other words, it doesn't works)

[Video del 05-11-2024 15:16:06.webm](https://github.com/user-attachments/assets/17a707bd-bc20-49d1-9bb9-c23dbc003d1c)

**Comment by Antoviscomi** [comment]
> in template mode there are no contents in tooltips modal popup (in other words, it doesn't works)

@gustry sorry for mistake in html mode it works but I doesn't seen the contents because of white color font

[Video del 05-11-2024 15:51:12.webm](https://github.com/user-attachments/assets/0521f158-2750-412c-bb64-4f1a41bdab56)

**Comment by Gustry** [comment]
> @Gustry sorry for mistake in html mode it works but I doesn't seen the contents because of white color font

This is because of your theme

**Comment by Antoviscomi** [comment]
You're right @Gustry in html it works as expected and also it's cusomizable with few html tags 

[Video del 05-11-2024 16:23:45.webm](https://github.com/user-attachments/assets/68296805-68af-4ea2-ae56-4eb112f64a41)

**Comment by josemvm** [comment]
i can confirm the problem, for a lizmap tooltip (i don´t try the qgis/html tooltip)

**Comment by Antoviscomi** [comment]
> i can confirm the problem, for a lizmap tooltip (i don´t try the qgis/html tooltip)

@josemvm it's not qgis/html tooltip but @Gustry meant the html model builder located in _Map Suggestions_ under Lizmap Plugin, using the second tab _HTML Model_ as in screenshot

![Schermata del 2024-11-05 17-38-05](https://github.com/user-attachments/assets/534be89d-ca57-4d05-b8ff-9ec7cdf1f35c)

**Comment by josemvm** [comment]
> @josemvm it's not qgis/html tooltip but @Gustry meants the html model builder located in Map Suggestions under Lizmap Plugin, using the second tab HTML Model as in screenshot

thanks @Antoviscomi

**Comment by guenterw** [comment]
I can also confirm this.
Just using the HTML template from the Lizmap plugin is enough to make it work.
![image](https://github.com/user-attachments/assets/6d850050-ae2b-409e-a787-16d3f8cfdfcb)

results in
![image](https://github.com/user-attachments/assets/03d87b7f-912f-43e6-9b8e-c63212806d80)

at least a workaround until the error is corrected

**Comment by Gustry** [comment]
> at least a workaround until the error is corrected

Yes, it's a workaround, but I would keep this HTML template. It's way more powerful, as you can write your own HTML and use QGIS expressions.

**Comment by josemvm** [comment]
hi,

we’ve confirmed that the issue occurs in several layers, but it is not related to the template being used. despite the error, the tooltip works perfectly in QGIS Desktop, which suggests that the data is correctly configured and accessible.

we’ve checked the QGIS Server, Lizmap, and PostgreSQL logs, as well as the Firefox developer console, but there are no error messages that could help us identify or trace the problem. this indicates that the issue might be related to how certain layers are being handled or rendered in Lizmap, though there are no clear signs from the server or client side.

thanks

**Comment by guenterw** [comment]
@josemvm thank you for your feedback.
When using the fields, the error also occurs in my current projects with version 3.9.1. When using the fields, the tooltip does not work.
It works perfectly with the HTML template.
Perhaps we should do away with the variant with the fields altogether?

**Comment by josemvm** [comment]
>It works perfectly with the HTML template.

i use the same template for all layers, but there are some layers where the error occurs and there's no way to trace it, because the same layer works fine in the QGIS tooltip as well as in Lizmap tables and popups.

**Comment by guenterw** [comment]
Oh, even with the HTML templates. Then I misunderstood.
So far, I've only seen the error with the version that has fields.

**Comment by mdouchin** [comment]
Fix with #6014 . It has been backported into 3.8,  3.9, future 3.10
